_source/_posts/2022-08-08-react-redux.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: blog_post
+title: Using Redux with React
+author: gabi-dombrowski
+by: contractor
+communities: [javascript]
+description: ""
+tags: [javascript, typescript, react, redux, usecontext, usestate, authentication]
+tweets:
+  - ""
+  - ""
+  - ""
+image:
+type: conversion
+canonical: https://auth0.com/blog/using-redux-with-react/
+---
+
+There are a myriad of state management options available for React. React provides the option of using the built-in [Context](https://reactjs.org/docs/context.html) for when you have a nested tree of components that share a state. There is also a built-in [useState hook](https://reactjs.org/docs/context.html) that will allow you to set local state for a component.
+
+For more complex scenarios where you might need a single source of truth that shared across large sections of your application and is frequently changing, you might want to consider using a more robust state management library.
+
+Let's compare when to use the useState hook vs. React Context vs. a more global state management solution like one of the most popular ones, [Redux](https://redux.js.org/).
+
+## Initial Setup
+
+We'll get started with the built-in React template for [Create React App](https://github.com/facebook/create-react-app) that uses Redux and Typescript. This includes the recommended Redux Toolkit and Redux's integration with React components.
+
+1. First, run:
+```bash
+npx create-react-app <YOUR_APP_NAME> --template redux-typescript
+```
+
+2. Then we'll add the Redux core by running:
+```bash
+npm add redux
```

Comment:
```suggestion
npm add redux@4.2
```

(or whatever version is correct. I also didn't test adding the version to this command so hopefully it works)